### DISCUSSION TITLE
How can I test IPA?

### DISCUSSION THREAD

**Discussion by kanisuki** [airsdk/Adobe-Runtime-Support#3003] [Q&A]
I made an ipa for AppStore with AIR.
I want to test this ipa.
Xcode can not take in the ipa made with AIR..
How can I test this ipa?

**Comment by ajwfrost** [comment]
Can you deploy it to a locally connected phone, in order to test it?
Either via AIR SDK:
```
/path/to/AIR/bin/adt -installApp -package filename.ipa
```
But in Xcode, if you go to Window -> Devices and Simulators, and select your connected phone, can you not click on the + under "Installed Apps" to browse for and install the IPA file?

thanks

**Reply by kanisuki** [reply]
Hello, ajwfrost.
Suppose my ipa name is "abcde.ipa".
I opened Xcode and clicked "Window" and choosed "Devices and Simulators"
and selected my connected iphone 12, but "Unable to install "abcde.ipa" "
was appeared. In the Details many lines appeared.

With AIR SDK,
How do I do with "/path/to/AIR/bin/adt -installApp -abcde.ipa"?

I connected my iphone12 to my Window PC and MAC PC.
I want to try several  simulators but some  tools need Xcode and others worked bad.

Thanks.

**Reply by ajwfrost** [reply]
If you're trying to install via AIR SDK / ADT, then open a command prompt or terminal window and change directory to the one that contains `abcde.ipa`. Then, either ensure the AIR SDK's "bin" folder is in your path, or preceed the "adt" command with its path.

Then to check that the device is connected properly, you can run
```
adt -devices -platform ios
```
Assuming you only have one device, it should then work if you do
```
adt -installApp -package abcde.ipa
```
If there are multiple devices, you can select the one to install it on via
```
adt -installApp -platform ios -device device-uuid -package abcde.ipa
```

Note that an IPA file that's created for a device (like ipa-app-store etc) can't then be installed on a simulator, you'd need to use a specific simulator build type for that.

If you want to screenshot the errors you're seeing in Xcode/Devices, we can see whether this shows a more fundamental problem that may be causing the issues here?

thanks

**Reply by M4rkY4** [reply]
I've always installed IPA simply by drag-and-dropping to a device in iTunes (on Windows) or Finder (macOS), assuming you just want to try your app on your iOS/iPadOS device and you use correct type of certificate. If "ipa for AppStore" means you've used your final/production/AppStore certificate, I'm not sure that you can test it on your personal device, isn't it only for uploading?

**Reply by pol2095** [reply]
for debugging an IPA, see [here](https://help.adobe.com/en_US/air/build/WS901d38e593cd1bac7b2281cc12cd6bced97-8000.html)

**Reply by kanisuki** [reply]
Thank you all. I am studying.

**Reply by kanisuki** [reply]
Hello, ajwfrost.
I sent Attach file of error.rtf (wordpad)
[error.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/13935529/error.zip)

**Reply by ajwfrost** [reply]
Sounds like your provisioning profile isn't set up to correspond to the app and/or device you're using:
> A valid provisioning profile for this executable was not found.

"for this executable" makes me think it's the app ID perhaps .. can you check in your provisioning profile (in the Apple developer account online) to see what application ID filtering is included in this? and then check that the application ID in your app descriptor XML file is matching it?

If you want, we can take a look at the IPA to check this, as that should include the profile and the descriptor file..

thanks

**Reply by kanisuki** [reply]
I understood what you told. I will read some web pages, and will try by myself.  If I have questions I will ask. 
Thanks.
